Updated Quarkus theme

adelasofia opened the pull request:
@jkbrock requested this PR that only contains the updated theme for the site. Unlike the other PRs ( https://github.com/quarkusio/quarkusio.github.io/pull/331 and https://github.com/quarkusio/quarkusio.github.io/pull/330 ) this does not contain the version announcement.

gsmet commented:
AFAICS, you used the same branch for the 2 PRs so I have the announcement in both. Could you adjust, please?

gsmet commented:
More specifically, the issue is with the homepage hero.

emmanuelbernard commented:
@gsmet I don't see the announcement when I build this PR

emmanuelbernard commented on _sass/layouts/guide.scss:
```diff
@@ -1,26 +1,39 @@
+      color: $link;
+    }
+    pre {
+      background-color: $white;
```
We lost syntax color for code in the guides. e.g. http://localhost:4000/guides/getting-started-guide#the-jax-rs-resources

emmanuelbernard replied:
Note also that the TIPS icon and maybe other icons don't look well on the gray background of guides. See right above http://localhost:4000/guides/getting-started-guide#using-injection

gsmet replied:
As for syntax highlighting I'm not sure it's a new issue, it never worked locally for me.

emmanuelbernard replied:
So it's a new issue, I can get syntax highlighting on the develop branch locally.

gsmet commented:
Either Adela pointed to another branch or I totally messed up this morning.

gsmet commented on _data/projectfooter.yaml:
```diff
@@ -17,25 +21,31 @@ links:
 
 more_links:
-  - title: Quarkus is proudly made of a number of projects including ...
+  - title: Quarkus is made of community projects
```
This change concerns me a bit as Mark L. specifically asked for it so that we know there are others. I don't mind shortening the text but we need a way to pass this message.

gsmet replied:
Note: it's not blocking, better focus on the CSS issues.

gsmet commented on _data/projectfooter.yaml:
```diff
@@ -17,25 +21,31 @@ links:
+  - title: Quarkus is made of community projects
     subfolderitems:
+      - page: Apache Camel
+        url: https://camel.apache.org
```
I think Apache Camel should be at the bottom, especially with the extensions being outside of the tree now. I don't see the value in sorting that alphabetically, better have it by importance.

gsmet commented on _data/projectfooter.yaml:
```diff
@@ -17,25 +21,31 @@ links:
+      - page: Thorntail
+        url: https://thorntail.io/
+      - page: WildFly
+        url: https://wildfly.org/
```
These two are definitely incorrect. Not sure about the third one.

gsmet replied:
@emmanuelbernard I think it's too early for this part. We don't really have any projects based on Quarkus to put there. Do you agree?

emmanuelbernard commented on _sass/includes/header-navigation.scss:
```diff
@@ -1,8 +1,22 @@
+body.guides,
```
In the list of guides http://localhost:4000/guides/
Look for the bottom of the page,

> When measuring the footprint of a Quarkus application, we measure Resident Set Size (RSS) and not the JVM heap size which is only a small part of the overall problem. The JVM not only allocates native memory for heap (-Xms, -Xmx) but also structures required by the jvm to run your application.

The element in parenthesis is code (marked class highlighter-rouge) and does not render well in the black background. I think we might need different asciidoc css depending on the light or dark background

emmanuelbernard commented on _sass/includes/header-navigation.scss:
```diff
@@ -1,8 +1,22 @@
+body.guides,
+body.post {
```
@FroMage check out http://localhost:4000/guides/hibernate-orm-guide#properties-to-refine-your-hibernate-orm-configuration on this PR
You will see that the config colors clash with the new light background for guides.

emmanuelbernard replied:
Also /guides/all-config is not in light background

emmanuelbernard commented on _includes/community-help-band.html:
```diff
@@ -0,0 +1,68 @@
+<div class="component-wrapper contributions-band">
```
In the /community page we lost:

- Quarkus QTips reference
- publications

Also I am not liking the translate in your language: that's a lot of work and infra from us that we don't have the bandwidth for.

jkbrock replied:
This is a content thing, so it's totally up to you guys if you want to remove it or not. @adelasofia is removing Translate.

We'll add in custom icons for QTips and Publications but they might come in a separate PR.

jkbrock replied:
@emmanuelbernard, are these icons and descriptions OK?
![Screen Shot 2019-11-04 at 3 22 21 PM](https://user-images.githubusercontent.com/10758508/68159243-02558300-ff17-11e9-9341-a8d657d287ff.png)

emmanuelbernard replied:
@jkbrock  The Quarkus _community_ publishes thoughtful...
The rest looks good.

emmanuelbernard replied:
I fixed the text

gsmet replied:
I was on it but you already fixed it, thanks!

gsmet commented on _includes/homepage-container_first-band.html:
```diff
@@ -0,0 +1,13 @@
+    </div>
+    <div class="width-12-12 block-content">
+      <p>Quarkus tailors your application for GraalVM and HotSpot. Amazingly fast boot time, incredibly low RSS memory (not just heap size!) offering near instant scale up and high density memory utilization in container orchestration platforms like Kubernetes. We use a technique we call compile time boot. <a href="/vision/container-first">Learn more.</a></p>
+      <pre class="highlightjs highlight"><code class="bash">$ ./my-native-java-rest-app
```
I agree that the previous terminal with dark grey background would be better. Especially since our syntax highlighter theme is configured for it.

jkbrock replied:
@adelasofia is reverting

gsmet commented on _data/projectfooter.yaml:
```diff
@@ -17,25 +21,31 @@ links:
-        url: https://camel.apache.org
+  - title: And used in so many more
+    subfolderitems:
+      - page: Thorntail
```
I can do this change, @adelasofia @jkbrock could you focus on the style ones? Thanks!

maxandersen commented on _includes/homepage-video-band.html:
```diff
@@ -0,0 +1,11 @@
+<div class="full-width-bg component homepage-content-band video-band">
+  <div class="grid-wrapper">
+    <div class="width-6-12 width-12-12-m video-container">
+      <iframe src="https://www.youtube.com/embed/hhHgurtI674" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
```
Its nice with a introductory video but the tone/content of it don't really send out a vide of "this project is interesting and new". 

I would postpone adding such content until we have a video with a bit more developer centric and to the point story.

emmanuelbernard replied:
I agree, I think this video is better suited for the upcoming RHT Quarkus website which will be more business minded.

maxandersen commented on _includes/homepage-container_first-band.html:
```diff
@@ -0,0 +1,13 @@
+      <p>Quarkus tailors your application for GraalVM and HotSpot. Amazingly fast boot time, incredibly low RSS memory (not just heap size!) offering near instant scale up and high density memory utilization in container orchestration platforms like Kubernetes. We use a technique we call compile time boot. <a href="/vision/container-first">Learn more.</a></p>
+      <pre class="highlightjs highlight"><code class="bash">$ ./my-native-java-rest-app
+Quarkus started in 0.008s</code></pre>
+      <img src="{{site.baseurl}}/assets/images/homepage_container_first.png">
```
The old style was much more catchy and recognizable - the new one is very "corporate" and my eye instinctually just gloss over it.

Could we keep the old terminal/matrix style but choose one of the quarkus colors instead ?

emmanuelbernard replied:
CC @jkbrock ^

cescoffier commented:
Configuration table are not very readable:

https://monosnap.com/file/AMutMvkvHMHX55e0PQZZyuFv0IwIpX

cescoffier commented:
New pictures: https://github.com/quarkusio/quarkus/pull/5177

adelasofia commented on _includes/homepage-container_first-band.html:
```diff
@@ -0,0 +1,13 @@
+      <p>Quarkus tailors your application for GraalVM and HotSpot. Amazingly fast boot time, incredibly low RSS memory (not just heap size!) offering near instant scale up and high density memory utilization in container orchestration platforms like Kubernetes. We use a technique we call compile time boot. <a href="/vision/container-first">Learn more.</a></p>
+      <pre class="highlightjs highlight"><code class="bash">$ ./my-native-java-rest-app
+Quarkus started in 0.008s</code></pre>
+      <img src="{{site.baseurl}}/assets/images/homepage_container_first.png">
```
@emmanuelbernard Jason is reworking the image so it can be more readable on smaller screens 👍

adelasofia commented:
@emmanuelbernard @maxandersen @gsmet @cescoffier 

I've pushed most of the changes that were requested (minus some of the content updates that were marked as not blockers). Here is a list of the changes in the new commit:

- Blog  and blog post pages: Updated blog icons to blue
- Blog post pages: Fixed footer type colors
- Blog post and Guides pages: Fixed link color on alternate background pages
- Blog post and Guides pages: On alternate background pages, changed color of vertical bar between the icon and the text in the admonitions
- Homepage: Added more space between content and image on the Container First section
- Homepage: Returned code snippets to have the old terminal style
- Community page: removed the “Translate into your language” item
- Removed homepage video section
- Updated logo (all white on homepage, colored logo on all other pages)
- Community page: added the Quarkus Qtips link (I need an icon to add the publications link, will create an issue for that)
- Guides page: fixed color of code snippet at the very bottom (the color was previously the same as the background so it was not readable)

adelasofia commented:
I've just pushed @jkbrock's updated graphic for the **Container First** section on the homepage.

gsmet commented:
FWIW, I already fixed the "Follow WildFly on Twitter" in Adela's branch.
It's Quarkus already.

Otherwise, I think it looks good.

On Mon, Nov 4, 2019 at 10:23 PM Jason Brock <notifications@github.com>
wrote:

> *@jkbrock* commented on this pull request.
> ------------------------------
>
> In _includes/community-help-band.html
> <https://github.com/quarkusio/quarkusio.github.io/pull/332#discussion_r342276573>
> :
>
> > @@ -0,0 +1,68 @@
> +<div class="component-wrapper contributions-band">
>
> @emmanuelbernard <https://github.com/emmanuelbernard>, are these icons
> and descriptions OK?
> [image: Screen Shot 2019-11-04 at 3 22 21 PM]
> <https://user-images.githubusercontent.com/10758508/68159243-02558300-ff17-11e9-9341-a8d657d287ff.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/quarkusio/quarkusio.github.io/pull/332?email_source=notifications&email_token=AAJYOBI3LKP23L6Z5LHITJ3QSCHEHA5CNFSM4JIPETKKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCKHRMLI#discussion_r342276573>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJYOBOOCLZ7D2N6BKLKV6TQSCHEHANCNFSM4JIPETKA>
> .
>

gsmet commented:
I iterated with Adela today and it looks good enough to be pushed. If you spot other issues, please open separate issues.

gsmet commented:
Thanks @adelasofia @jkbrock !
